### PR TITLE
Support per-hook stages

### LINKED ISSUE #110: Support specifying stages for individual hooks
Continuation of #49 

Currently, it's possible to specify `default_stages` for all hooks, which also configures which stages hooks are installed for.

Via the `raw` attribute on an individual hook, it's possible to specify `stages` for a single hook, though if any stage specified is not in the `default_stages`, the hook will not be installed.

So I suppose there are two aspects to this:

1. Add a `stages` config property on the hook level
2. Update `install_stages` to install both default stages (per current behaviour) and any specified in individual hooks

I'm happy to work on the implementation for this but it might be a few days before I get the time to do so, so I figured I'd post here first in case anyone has any comments.

### PR DESCRIPTION
**Pull request by sd234678**
Closes #110

### CODE REVIEW

**Review by roberth — CHANGES_REQUESTED** [review]

**Comment by sd234678** [comment]
Hey @roberth, I don't suppose you could re-review this?

File: modules/pre-commit.nix [hunk 1]
```diff
@@ -117,12 +117,18 @@ let
                 description = "Whether to pass filenames as arguments to the entry point.";
                 default = true;
               };
+            stages =
+              mkOption {
+                type = types.listOf types.str;
+                description = "Confines the hook to run at a particular stage.";
+                default = cfg.default_stages;
```
Comment (roberth) [inline modules/pre-commit.nix:124]:
```suggestion
                defaultText = (lib.literalExpression or lib.literalExample) "default_stages";
```

**Review by roberth — COMMENTED** [review]
Otherwise lgtm

File: modules/pre-commit.nix (hunk 1 above)
Comment (sd234678) [reply modules/pre-commit.nix]: Thanks, I've updated that now to match

Comment (sd234678) [reply modules/pre-commit.nix:124]: Removing the value for `default` broke the build as stages was left without a value, but I've added the `defaultText` in addition. Was that what you were intending?

Comment (roberth) [reply modules/pre-commit.nix:124]: It is; I just used the suggestion feature wrong. Sorry about that.

Comment (sd234678) [reply modules/pre-commit.nix:124]: No worries, thanks for your help 😄

**Comment by sd234678** [comment]
Hey @roberth, this all seems to be working okay now

**Comment by roberth** [comment]
@domenkozar ^

**Pull request merged**